### PR TITLE
Add opt-in box for facilitated dialogues on Frontpage

### PR DESCRIPTION
**Pull request by lagerros**
This PR includes: 
* adding a component for asking the user if we can facilitate a dialogue for them
* adding a field to the users schema allow user to hide the component
* make migration for the new field
* add A/B test for different phrasing of the prompt
* when a user opts in to facilitation, ping a webhook that messages us on slack. (Deliberately putting this on client following discussion with rafe and robert because eh, yolo.) Also creating an analytics event

(Screenshots are from a version without the latest phrasing)

image

image

### CODE REVIEW

File: packages/lesswrong/components/dialogues/DialoguesList.tsx [hunk 1]
```diff
@@ -1,12 +1,138 @@
...
+
+  closeIcon: { 
+    position: 'absolute', 
+    right: '8px',
+    top: '8px',
+    padding: '2px' 
+  }
+});
+
+const OptInComponent = ({ classes, currentUser, setShowOptIn }: { classes: ClassesType, currentUser: UsersCurrent | null, setShowOptIn: React.Dispatch<React.SetStateAction<boolean>> }) => {
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialoguesList.tsx:58]: We don't suffix component names with `Component`, generally.  I'd also make the name more descriptive (i.e. `DialogueFacilitationPrompt` or something).

Also, if you're going to be passing in `currentUser` from the parent, it should probably not be nullable - it doesn't really make sense to show this to logged-out users.

**Comment by b0b3rt** [comment]
In general something about the text styling looks pretty off.  The bolding is particulary bad but I think it's also the wrong font.

**Comment by lagerros** [comment]
> In general something about the text styling looks pretty off. The bolding is particulary bad but I think it's also the wrong font.

It's Gill Sans from commentBodyStyles. Are we using multiple different Gill Sans?

File: packages/lesswrong/lib/collections/users/schema.ts [hunk 2]
```diff
@@ -1392,6 +1392,17 @@ const schema: SchemaType<DbUser> = {
     hidden: !dialoguesEnabled,
   },
 
+  hideDialogueFacilitation: {
+    type: Boolean,
+    canRead: [userOwns, 'sunshineRegiment', 'admins'],
+    canCreate: ['members'],
+    canUpdate: [userOwns, 'sunshineRegiment', 'admins'],
+    optional: true,
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collections/users/schema.ts:1400]: This should have an explicit `nullable: false`, and a `...schemaDefaultValue(false)`

Comment (lagerros) [reply packages/lesswrong/lib/collections/users/schema.ts:1400]: fixed

Comment (b0b3rt) [reply packages/lesswrong/lib/collections/users/schema.ts:1400]: Unfortunately you will then also need to delete the generated migration file, as well as undo any changes to `accepted_schema.sql` and `schema_changelog.json` (you can run `git checkout origin/master -- schema/accepted_schema.sql schema/schema_changelog.json` to overwrite whatever you have with whatever's latest in master).  Then run `yarn makemigrations` again, update the generated migration file, etc, etc.

Comment (lagerros) [reply packages/lesswrong/lib/collections/users/schema.ts:1400]: Makes sense. Now did all these things again!

**Comment by lagerros** [comment]
@b0b3rt styling improvements 

image

image

**Review by b0b3rt — APPROVED** [review]

**Pull request merged**